### PR TITLE
Move heartbeat to standalone exporter

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/heartbeat/HeartbeatExporter.java [hunk 1]
```diff
@@ -132,8 +132,7 @@ TelemetryItem gatherData() {
       numHealthy += payload.isHealthy() ? 0 : 1;
     }
     MetricTelemetryBuilder telemetryBuilder =
-        telemetryClient.newMetricTelemetryBuilder(HEARTBEAT_SYNTHETIC_METRIC_NAME, numHealthy);
-
+        MetricTelemetryBuilder.create(HEARTBEAT_SYNTHETIC_METRIC_NAME, numHealthy);
```
[the remark this answers is not in the document]
Comment (heyams) [reply agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/heartbeat/HeartbeatExporter.java:145]: good catch.

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java [hunk 2]
```diff
@@ -153,7 +156,17 @@ public void customize(AutoConfigurationCustomizer autoConfiguration) {
 
     // interval longer than 15 minutes is not allowed since we use this data for usage telemetry
     long intervalSeconds = Math.min(config.heartbeat.intervalSeconds, MINUTES.toSeconds(15));
-    HeartBeatProvider.start(intervalSeconds, telemetryClient);
+    Consumer<List<TelemetryItem>> telemetryItemsConsumer =
+        telemetryItems -> {
+          for (TelemetryItem telemetryItem : telemetryItems) {
+            TelemetryObservers.INSTANCE
+                .getObservers()
+                .forEach(consumer -> consumer.accept(telemetryItem));
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java:164]: this accept does the populateDefaults @trask? i might need to remove telemetryinitializer in HeartbeatExporter.

Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java:164]: no it doesn't. please ignore this. it was passed to MetricDataMapper. we need to keep it

File: agent/instrumentation/micrometer-1.0/src/test/groovy/MicrometerTest.groovy [hunk 3]
```diff
@@ -226,6 +226,6 @@ class MicrometerTest extends AgentInstrumentationSpecification {
     def measurements = delegate.getMeasurements().stream()
       .filter({ it.name == name && it.value != 0 })
       .collect(Collectors.toList())
-    return measurements.get(measurements.size() - 1)
+    return measurements.size() > 0 ? measurements.get(measurements.size() - 1) : null;
```
Comment (trask) [inline agent/instrumentation/micrometer-1.0/src/test/groovy/MicrometerTest.groovy:229]: I think returning `null` will end up throwing NullPointerException above on line 219

**Review by trask — APPROVED** [review]

File: agent/instrumentation/micrometer-1.0/src/test/groovy/MicrometerTest.groovy (hunk 3 above)
Comment (heyams) [reply agent/instrumentation/micrometer-1.0/src/test/groovy/MicrometerTest.groovy:229]: will revert it now after skipping cache successfully.

Comment (heyams) [reply agent/instrumentation/micrometer-1.0/src/test/groovy/MicrometerTest.groovy:229]: or adding null check everywhere?

**Pull request merged**